dfkernel/utils.py

```diff
@@ -110,18 +116,62 @@ def visit_Name(self, node):
                 self.scope[-1].add(node.id)
             elif isinstance(node.ctx, ast.Del):
                 self.scope[-1].discard(node.id)
-            elif (isinstance(node.ctx, ast.Load) and
+            elif (not reversion) and (isinstance(node.ctx, ast.Load) and
                     all(node.id not in s for s in self.scope) and
                     dataflow_state.has_external_link(node.id, execution_count)):
                 cell_id = dataflow_state.get_external_link(node.id, execution_count)
+                if not (display_code_parsed and output_tags.get(node.id) and len(output_tags[node.id]) == 1 and cell_id in output_tags[node.id]):
+                    ref = DataflowRef(
+                        start_pos=(node.lineno, node.col_offset),
+                        end_pos=(node.end_lineno, node.end_col_offset),
+                        name=node.id,
+                        cell_id=cell_id
+                    )
+                    self.updates.append(ref)
+                
+            elif (not reversion) and (isinstance(node.ctx, ast.Load) and
```

Comment:
There seems to be repetition of conditions from line 119 here?

Reply:
refactored code to avoid repetition